### ISSUE TITLE
Shades

### ISSUE THREAD

**Issue by Mikelawry214** [node-alarm-dot-com/homebridge-node-alarm-dot-com#95]
Alarmdotcom has now added the ability to utilize Somfy Shades on their platform. The Somfy shades are still connected through zwave but show in a different area as shade rather than lights.

Can we get this added as an accessory as well. I added a few photos of the item in the app.

![image](https://user-images.githubusercontent.com/101213951/157316041-0289fe10-cfb0-4965-973c-cda90a8168c5.jpeg)

![image](https://user-images.githubusercontent.com/101213951/157316118-a5b61c57-c203-445f-be76-3a97c36c163c.jpeg)

![image](https://user-images.githubusercontent.com/101213951/157316188-403bc9ee-ca44-48c7-945f-559766bbf6f8.jpeg)

**Comment by ngori** [comment]
@Mikelawry214 This enhancement shouldn't be too difficult to add. We would just need to capture the API calls for the different commands specific for shades to be able to add it.  My guess is it's similar to a dimmer (i.e.,  Up/Down/% vs On/Off/%).  A lot of the dimmer code might be reusable here.   When I added the garage door support I used the Chrome  Developer Tools to capture the network activity of the URL Request on the alarm.com website.  Garage doors even more basic though.  Once you can capture the URL Request, Request Headers, Payload, etc. it's pretty straight forward.

**Comment by Mikelawry214** [comment]
> @Mikelawry214 This enhancement shouldn't be too difficult to add. We would just need to capture the API calls for the different commands specific for shades to be able to add it.  My guess is it's similar to a dimmer (i.e.,  Up/Down/% vs On/Off/%).  A lot of the dimmer code might be reusable here.   When I added the garage door support I used the Chrome  Developer Tools to capture the network activity of the URL Request on the alarm.com website.  Garage doors even more basic though.  Once you can capture the URL Request, Request Headers, Payload, etc. it's pretty straight forward.
> 
> 
> 
> 
> 
> 
> 
> 
> 
>   

Yes it is similar to a dimmer switch. If the ability to add is available, what would I do on my end to update it to see if within my accessories. Thanks for your help

**Comment by Mikelawry214** [comment]
> @Mikelawry214 This enhancement shouldn't be too difficult to add. We would just need to capture the API calls for the different commands specific for shades to be able to add it. My guess is it's similar to a dimmer (i.e., Up/Down/% vs On/Off/%). A lot of the dimmer code might be reusable here. When I added the garage door support I used the Chrome Developer Tools to capture the network activity of the URL Request on the alarm.com website. Garage doors even more basic though. Once you can capture the URL Request, Request Headers, Payload, etc. it's pretty straight forward.

I found the settings to be very similar obviously with a different identifier 

Not sure what I would do next.

**Comment by ngori** [comment]
@Mikelawry214  I can't take it on right now due to other family/work commitments but in general the process would be capture the api calls.  Write the new methods for up, down, status, in this package and the node-dot-alarm-com package, create a pull request merge that into a beta and then test.

**Comment by Mikelawry214** [comment]
Thank you. I totally understand.

Honestly I have no idea about coding.

**Comment by Mikelawry214** [comment]
Anyone willing to take on this. Willing to donate money for your work

**Comment by ngori** [comment]
I may have a bit of time coming up.  There is a discord server dedicated to this plugin

https://discord.com/channels/432663330281226270/742182845056942210

I can describe the steps I'd need to implement shades over there.

**Comment by Mikelowry1** [comment]
Sure just let me know

**Comment by ngori** [comment]
The channel on the discord is #node-alarm-dot-com

**Comment by Mikelowry1** [comment]
Unfortunately that link isn't working for me on discord

**Comment by ngori** [comment]
Sorry.  Try this https://discord.gg/Z8jmyvb

**Comment by ngori** [comment]
I'm working on this one over the next 2-3 weeks with @Mikelowry1 to get shades implemented.

**Comment by Stugatza** [comment]
hi all, any updates on shade integration?  would be great to have.

**Comment by chase9** [comment]
Since I don’t have shades I would need someone to contribute a PR based on the API calls they see in the web portal.ChaseOn Oct 6, 2023, at 4:12 PM, Stugatza ***@***.***> wrote:﻿
hi all, any updates on shade integration?  would be great to have.

**Comment by chase9** [comment]
Hey @ngori and @Stugatza, I've been working on refactoring the code bases and I'm more comfortable implementing new devices now. I'd love to help get this into the plugin. Was any progress made on shade support?

**Comment by ngori** [comment]
None from me. I switched ADC providers after a move and now run most everything through Home Assistant.

**Comment by chase9** [comment]
OK, thanks for responding!

Thank you,
Chase
________________________________

**Comment by Stugatza** [comment]
thanks all, i do have a sonoma integration, but no other blinds/screens
that i can test with

**Comment by chase9** [comment]
Is that integrated with Alarm.com? I would just need you to relay some of the calls that are made when you try to control them with the website.
